arangod/RestHandler/RestAdminClusterHandler.cpp

```diff
@@ -2902,3 +2910,55 @@ RestAdminClusterHandler::collectRebalanceInformation(
 
   return p;
 }
+
+RestAdminClusterHandler::FutureVoid
+RestAdminClusterHandler::handleVPackSortMigration(
+    std::string const& subCommand) {
+  // First we do the authentication: We only allow superuser access, since
+  // this is a critical migration operation:
+  if (ExecContext::isAuthEnabled() && !ExecContext::current().isSuperuser()) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_FORBIDDEN,
+                  "only superusers may run vpack index migration");
+    co_return;
+  }
+
+  // First check methods:
+  if (!((request()->requestType() == rest::RequestType::GET &&
+         subCommand == VPackSortMigrationCheck) ||
+        (request()->requestType() == rest::RequestType::PUT &&
+         subCommand == VPackSortMigrationMigrate))) {
+    generateError(rest::ResponseCode::METHOD_NOT_ALLOWED,
+                  TRI_ERROR_HTTP_METHOD_NOT_ALLOWED);
+    co_return;
+  }
+
+  // We behave differently on a coordinator and on a single server,
+  // dbserver or agent.
+  // On a coordinator, we basically implement a trampoline to all dbservers.
+  // On the other instance types, we implement the actual checking and
+  // migration logic. Agents do not have to be checked since they do not
+  // use VPack indexes.
```

Comment:
I don't think that this statement is true.
Agents will use vpack indexes, too, at least for the system collections that are present on them, e.g. `_users`.
It is true though that we do not index numeric vpack values on agents, but we use vpack indexes there.

Reply:
Just checked 3.12 (startLocalCluster.sh) and 3.11.10 and my agent did not have a single system collection. Same 3.11. And the collections used by the agent do not use indexes at all.